From af54b8c12284af3a3a0a0bf5475c789352a76297 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Thu, 1 Jun 2023 12:54:06 +0530
Subject: [PATCH] Move mediatranscoding type to public

Move type to public so that it can be vendor customized. This
can be necessary if (for example) the gralloc/gpu same-process-HAL
requires additional permissions.

Bug: 199581284
Test: build
Change-Id: I61a5a3ad96112d4293fd4bf6d55f939c974643ce
(cherry picked from commit 1b2a0b4dab564e11cc04b3fcf9d735cb956239ff)
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>
---
 prebuilts/api/32.0/private/mediatranscoding.te | 2 +-
 prebuilts/api/32.0/public/mediatranscoding.te  | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)
 create mode 100644 prebuilts/api/32.0/public/mediatranscoding.te

diff --git a/prebuilts/api/32.0/private/mediatranscoding.te b/prebuilts/api/32.0/private/mediatranscoding.te
index 2a43cf9b5..ad9777862 100644
--- a/prebuilts/api/32.0/private/mediatranscoding.te
+++ b/prebuilts/api/32.0/private/mediatranscoding.te
@@ -1,5 +1,4 @@
 # mediatranscoding - daemon for transcoding video and image.
-type mediatranscoding, domain;
 type mediatranscoding_exec, system_file_type, exec_type, file_type;
 type mediatranscoding_tmpfs, file_type;
 typeattribute mediatranscoding coredomain;
@@ -19,6 +18,7 @@ hal_client_domain(mediatranscoding, hal_graphics_allocator)
 hal_client_domain(mediatranscoding, hal_configstore)
 hal_client_domain(mediatranscoding, hal_omx)
 hal_client_domain(mediatranscoding, hal_codec2)
+hal_client_domain(mediatranscoding, hal_allocator)
 
 allow mediatranscoding mediaserver_service:service_manager find;
 allow mediatranscoding mediametrics_service:service_manager find;
diff --git a/prebuilts/api/32.0/public/mediatranscoding.te b/prebuilts/api/32.0/public/mediatranscoding.te
new file mode 100644
index 000000000..420d03865
--- /dev/null
+++ b/prebuilts/api/32.0/public/mediatranscoding.te
@@ -0,0 +1 @@
+type mediatranscoding, domain;
-- 
2.17.1

